src/components/cc-popover/cc-popover.js

```diff
@@ -0,0 +1,217 @@
+    super();
+
+    /** @type {string|null} Sets the accessibleName property of the underlying `cc-button` element. CAUTION: The accessible name should always start with the visible text if there is one. For instance "add to estimation - NodeJS XS" */
+    this.accessibleName = null;
+
+    /** @type {boolean} Whether the button text should be hidden. */
+    this.hideText = false;
+
+    /** @type {IconModel|null} Sets the button icon. */
+    this.icon = null;
+
+    /** @type {boolean} Whether the popover is opened */
+    this.isOpen = false;
+
+    /** @type {PopoverPosition} Sets the position of the popover relative to the `button` element. */
+    this.position = 'bottom-left';
+
+    /** @type {Ref<HTMLElement>} */
+    this._contentRef = createRef();
+
+    /** @type {Ref<HTMLElement>} */
+    this._buttonRef = createRef();
+
+    this._onOutsideClickHandler = new EventHandler(window, 'click', (event) => {
+      const contentElement = this._contentRef.value;
+      if (contentElement != null && !event.composedPath().includes(contentElement)) {
+        this.close();
+      }
+    });
+    this._onEscapeKeyHandler = new EventHandler(this, 'keydown', (event) => {
+      if (event.key === 'Escape') {
+        this.close();
+      }
+    });
+  }
+
+  /**
+   * Opens the popover.
+   */
+  open () {
+    if (!this.isOpen) {
+      this.isOpen = true;
+      dispatchCustomEvent(this, 'open');
+    }
+  }
+
+  /**
+   * Closes the popover.
+   */
+  close () {
+    if (this.isOpen) {
+      this.isOpen = false;
+      this._buttonRef.value?.focus();
+      dispatchCustomEvent(this, 'close');
+    }
+  }
+
+  /**
+   * Toggle the popover display.
+   */
+  toggle () {
+    if (this.isOpen) {
+      this.close();
+    }
+    else {
+      this.open();
+    }
+  }
+
+  updated (_changedProperties) {
+    if (_changedProperties.has('isOpen')) {
+      if (this.isOpen) {
+        this._onOutsideClickHandler.connect();
+        this._onEscapeKeyHandler.connect();
+      }
+      else {
+        this._onOutsideClickHandler.disconnect();
+        this._onEscapeKeyHandler.disconnect();
+      }
+    }
+  }
+
+  disconnectedCallback () {
+    super.disconnectedCallback();
+    this._onOutsideClickHandler.disconnect();
+    this._onEscapeKeyHandler.disconnect();
+  }
+
+  render () {
+    return html`
+      <cc-button
+        ${ref(this._buttonRef)}
+        aria-expanded=${this.isOpen ? 'true' : 'false'}
+        @cc-button:click=${this.toggle}
+        .accessibleName=${this.accessibleName}
+        ?hide-text=${this.hideText} 
+        .icon=${this.icon}
+      >
+        <slot name="button-text"></slot>
+      </cc-button>
+
+      ${this.isOpen ? html`
+        <div class="content ${this.position.replace('-', ' ')}" ${ref(this._contentRef)}>
+          <slot></slot>
+        </div>
+      ` : null}
+    `;
+  }
+
+  static get styles () {
+    return [
+      // language=CSS
+      css`
+        :host {
+          position: relative;
```

Comment:
Nice catch! But I'm not sure if I would make this change. :thinking: 

Your suggestion will slightly complicate the DOM: it raises the question of how much effort we put in preventing wrong usage of the component regarding the cost of the potential trade-off?

Reply:
Two very good feedbacks :muscle:

@florian-sanders-cc Technically, the behaviour would only break if someone puts `position: static`, this is very unlikely. It would still work with `absolute` or `fixed`.

@roberttran-cc Seems like the DOM would only get a `div` wrapper. I don't see this as complicated.

I'm 51% leaning for the let's keep it like this.

Reply:
I'll go for a `div` wrapper